src/admin/bundles/bundle-settings.controller.js

```diff
@@ -0,0 +1,66 @@
+(function(){
+	'use strict';
+
+	angular.module('motech-admin')
+		.controller('BundleSettingsController', controller);
+
+	controller.$inject = ['$scope', '$stateParams', '$http', 'BundleSettingsFactory', 'BundleRawSettingsFactory', 'LoadingModal', 'ModalFactory', 'BundlesFactory'];
+	function controller($scope, $stateParams, $http, BundleSettingsFactory, BundleRawSettingsFactory, LoadingModal, ModalFactory, BundlesFactory){
+
+        var restartBundleHandler = function () {
+            $scope.module.$restart();
+        };
+
+        $scope.module = BundlesFactory.get({ bundleId:$stateParams.bundleId });
+
+        $scope.moduleSettings = BundleSettingsFactory.query({ bundleId:$stateParams.bundleId });
+
+        $scope.rawFiles = BundleRawSettingsFactory.query({ bundleId:$stateParams.bundleId });
+
+        $scope.saveSettings = function (mSettings, doRestart) {
+            var successHandler;
+            if (doRestart === true) {
+                successHandler = restartBundleHandler;
+            } else {
+                successHandler = function () {
+                    LoadingModal.close();
+                    ModalFactory.showSuccessAlert('admin.settings.saved');
+                };
+            }
+
+            LoadingModal.open();
+
+            mSettings.$save({bundleId:$scope.module.bundleId}, successHandler,
+                function(response) {
+                    LoadingModal.close();
+                    ModalFactory.showErrorAlertWithResponse('admin.settings.error', 'admin.error', response);
+                }
+            );
+        };
+
+        $scope.uploadRaw = function (filename, doRestart) {
+            var successHandler,
+            id = '#_raw_' + filename.replace('.', '\\.');
+
+            if (doRestart === true) {
+                successHandler = restartBundleHandler;
+            } else {
+                successHandler = function () {
+                    LoadingModal.close();
+                    ModalFactory.showSuccessAlert('admin.settings.saved');
+                };
+            }
+
+            LoadingModal.open();
+
+            $(id).ajaxSubmit({
```

Comment:
As suggested in [PR#39](https://github.com/motech/motech-ui/pull/39) we should not use JQueryFormPlugin as it is rather unsupported.